### PR TITLE
add CONTRIBUTING.md first draft

### PR DESCRIPTION
**Pull request by Flet**
Initial shot at this... Still uncertain of what tasks are required, but we can update as necessary.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+Creating a New Example Implementation
+============
+
+Getting Started
+---
+1. [Fork this repo](https://github.com/ericelliott/todotasks/fork)
+
+2. clone your fork and copy the `base` files
```
Comment (ericelliott) [inline CONTRIBUTING.md]: We really need to make a script to make this easier. See https://github.com/ericelliott/todotasks/issues/12.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    # install dependencies
+    cd todotasks/implementations/<your-build-process-name>
+    npm install
+    ```
+
+3. Implement all the tasks listed below in your selected build process and link them to npm scripts of the same name  in `package.json`.
+
+4. Update the `README.md` with detailed information about your task runner. See below for details.
+
+5. Verify that your example covers all the below tasks by running this helper script:
```
Comment (ericelliott) [inline CONTRIBUTING.md]: Maybe we should land the implementation for https://github.com/ericelliott/todotasks/issues/10 before we merge this.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+---
+Your implementation example must implement all of the tasks below. Each task should be mapped to an `npm` script of the same name.
+
+Example `package.json`:
+```
+{
+  "name": "todotasks-template-gulp",
+  "version": "1.0.0",
+  "description": "",
+  "scripts": {
```
Comment (ericelliott) [inline CONTRIBUTING.md]: Add `test`.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Flet) [inline CONTRIBUTING.md]: Indeed!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+Delete all files in the `dist` directory.
+
+**lint**
+Run `eslint` on all js files in the `js` directory.
+
+**css**
+Use node-sass to compile `bower_components/todomvc-common/*.css`and output to `dist/app.css`.
+
+**uglify**
+Take all JavaScript  from `/src/plainjs/*`and generate a single concatenated+uglify'd file at `dist/ugify.js`. Please use *uglfy* for this task.
```
Comment (ericelliott) [inline CONTRIBUTING.md]: I'm using `minifyify` to preserve sourcemaps, but I can change that. Also, why not call it `dist/app.js`.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+If you need to write the concatenated file to the filesystem before uglifying, write it to `dist/concat.js`.
+
+**browserify**
+Use browserify to generate `dist/browserify.js` from the file `js/app.js`. This should also use the `minifyify` plugin and generate `dist/bundle.map.json`.
+
+**build**
+This task should perform all of the above tasks in order. Note that this should use the implementations own internal task references (do not use the `*`npm scripts).
```
Comment (ericelliott) [inline CONTRIBUTING.md]: Should be:

* lint
* clean
* compile / bundle
* test

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+**dev**
+This is the most complicated task. It should watch `src/plainjs` for changes and when a change is detected do the following:
```
Comment (ericelliott) [inline CONTRIBUTING.md]: I believe this dir is just called `js` in this repo.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    - run eslint against all files in`src/plainjs` and display any errors to stdout. Linting errors should *not* stop this task.
+    - rebuild a concatenated+uglify'd file at `dist/ugify.js`. To make this task fast, strive to rebuild only what needs to be rebuilt.
```
Comment (ericelliott) [inline CONTRIBUTING.md]: Again, I'm not sure I see the point of the filename change here...

**Comment by ericelliott** [comment]
Overall, great start! Thanks!

:+1:

File: CONTRIBUTING.md (hunk 4 above)
Comment (Flet) [inline CONTRIBUTING.md]: Caught a copy/paste from the old one. Will make it `dist/app.js`

Is it important to you to have sourcemaps extracted as part of this task?

Comment (ericelliott) [inline CONTRIBUTING.md]: I think extracted source maps is the only thing that really makes sense for large projects in prod, but I'm willing to drop that consideration. What are your thoughts?

**Comment by Flet** [comment]
Thanks for the quick feedback! I will fix my typos and filenames.

**Comment by Flet** [comment]
Should sass compilation be part of compile then?

**Comment by ericelliott** [comment]
Yes, sass compilation should be part of compile. We may also want to call that out as a required task. See [compile-styles](https://github.com/ericelliott/todotasks/blob/master/implementations/npm/package.json#L10) from the npm example.

**Comment by ericelliott** [comment]
Will you have time to address the issues soon?

**Comment by Flet** [comment]
Updates made... I removed the bundle map stuff for now... since we're asking them to use browserify, it will probably be very similar with each implementation anyway.

Please take another look at consider merging.

**Comment by ericelliott** [comment]
Thanks!

**Pull request merged**